### DISCUSSION TITLE
How to add extensions into electron app without extracting(app.asar modifications)?

### DISCUSSION THREAD

**Discussion by Jenifer-TheCoder** [samuelmaddock/electron-browser-shell#24] [Q&A] [answered]
Hi there ,special mention to @samuelmaddock 

I am trying to add custom extension into any electron application to monitor few DOM-UI properties.

- The way suggested in electron-browser-shell project seems to be,adding below line in main process(Main.js):

`extensions.addTab(browserWindow.webContents, browserWindow)`

- That is possible only by extracting app.asar file and repackaging it.
- Is there a way to do without extracting the asar file?
- If extracting is the only way, is it possible by some means of preload script so that main process script to be unmodified.
- The purpose here to have chrome-extension getting loaded into electron app and does the required job. Since it is client specific electron app, they might not want us to modify their asar files.

**Comment by samuelmaddock** [comment, marked as answer]
I don't see it being possible to add support for `electron-chrome-extensions` without the app author installing and configuring it themselves. The main process script will always need to be modified to initialize the [`Extensions`](https://github.com/samuelmaddock/electron-browser-shell/tree/master/packages/electron-chrome-extensions#class-extensions) class.

Also, since you mentioned this is for a client, make sure they've reviewed [this project's license](https://github.com/samuelmaddock/electron-browser-shell/tree/master/packages/electron-chrome-extensions#license) as it it will affect any application that integrates with `electron-chrome-extensions`.

**Reply by Jenifer-TheCoder** [reply]
Yeah sure, will keep you updated!!
